blog/_src/posts/2017-01-02-fall-2016-pl-junior-retrospective.md

```diff
@@ -0,0 +1,119 @@
+    Title: Fall 2016 PL Junior Retrospective
+    Date: 2017-01-02T16:39:37
+    Tags: PL Junior
+
+The Programming Language Seminar, Junior (or “PL Junior”), is a
+seminar for junior students to learn and discuss topics at a pace more
+suitable to our background. This semester, we decided to study
+dependent types. We chose this topic because 1) working from the [TAPL]
+[1] presentation of type systems, dependent types are a step up in
+difficulty (excepting F-omega-sub), and 2) they represent a
+significant increase in the reasoning power of types over programs.
+
+There was a preference for learning how to implement a dependent type
+system, instead of spending a significant amount of time reading
+papers, especially dense type-theoretic papers suggested by [posts]
+[4] like [these][5]. So we followed the [pi-for-all] [2] lecture
+series given by Stephanie Weirich at [OPLSS] [9], which focuses on
+implementing a simple dependently-typed programming language.
+
+After the pi-for-all lectures, we read chapter two of Edwin Brady’s
+[dissertation on implementing dependently typed languages] [3]. The
+thesis includes a relatively approachable introduction to TT, the core
+dependent type theory of Epigram.
+
+Along the way, we became sidetracked by [Girard’s paradox] [6]. In the
+first pi-for-all lecture, we came across the Type-in-Type rule. (In a
+dependent type system the term and the type languages are the same.
+However, we still need to distinguish what is a “program” and what is
+a “type,” for instance, so that we can determine that the annotation
+of a function’s argument is valid. So a construct in the term language
+is Type, which is meant to describe those things that are valid in
+programs where we expect to find a type). In the lecture, this
+prompted the comment that this (“of course”) makes our system
+inconsistent as a logic, but there was no further elaboration, and we
+could not figure out how to use this fact to show inconsistency.
+
+It turns out the reason Type-in-Type is inconsistent is quite
+complicated. It is explained in a [paper] [7] that we had difficulty
```

Comment:
I would read that post, but as we tried to get at I don't understand it well enough to give more than the hand-wavey explanation here.